File: Source/Java/RiceDoctor/src/main/java/vn/edu/uit/server/Response.java
package vn.edu.uit.server;

import com.google.gson.annotations.SerializedName;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import vn.edu.uit.shared.Buildable;

import java.util.HashMap;
import java.util.Map;

public class Response {

    @NotNull
    @SerializedName("Status")
    private final ResponseType status;

    @Nullable
    @SerializedName("Data")
    private final Map<String, Object> data;

    @Nullable
    @SerializedName("Message")
    private final String message;

    Response(@NotNull final Builder builder) {
        status = builder.status;
        data = builder.data;
        message = builder.message;
    }

    public static class Builder implements Buildable<Response> {

        @NotNull
        private final ResponseType status;

        @Nullable
        private Map<String, Object> data;

        @Nullable
        private String message;

        public Builder(@NotNull final ResponseType status) {
            this.status = status;
        }

        @NotNull
        public Builder data(@NotNull final String key, @Nullable final Object value) {
            if (data == null) data = new HashMap<>();
            data.put(key, value);
            return this;
        }

        @NotNull
        public Builder message(@NotNull final String message) {
            this.message = message;
            return this;
        }

        @Override
        @NotNull
        public Response build() {
            return new Response(this);
        }
    }
}

File: Source/C#/RiceDoctor/RiceDoctor.OntologyManager/Client/Response.cs
using System.Collections.Generic;
using JetBrains.Annotations;
using RiceDoctor.Shared;

namespace RiceDoctor.OntologyManager
{
    public class Response
    {
        public Response(ResponseType status, [NotNull] IReadOnlyDictionary<string, object> data)
        {
            Status = status;
            Data = data;
        }

        public ResponseType Status { get; }

        [CanBeNull]
        public IReadOnlyDictionary<string, object> Data { get; }

        [CanBeNull]
        public string Message { get; set; }

        public static Response Deserialize([NotNull] string json)
        {
            Check.NotEmpty(json, nameof(json));

            return JsonConvert.Deserialize<Response>(json);
        }
    }
}